### PR TITLE
Station enum flags

### PR DESCRIPTION
**Pull request by leslieyip02**
Addresses #1757 

Implemented enum flags for `StationFlags`. 
Created enum classes for `StationCargoStatsFlags` and `CatchmentFlags` (previously unnamed).

### CODE REVIEW

File: src/OpenLoco/src/Station.h [hunk 1]
```diff
@@ -51,24 +62,34 @@ namespace OpenLoco
...
     struct CargoSearchState;
 
+    enum class CatchmentFlags : uint8_t
+    {
+        none = 0U,
+        flag0 = 0U,
+        flag1 = 1U,
+    };
+    OPENLOCO_ENABLE_ENUM_OPERATORS(CatchmentFlags);
+
```
Comment (leslieyip02) [inline src/OpenLoco/src/Station.h:92]: https://github.com/OpenLoco/OpenLoco/blob/59cee31ad175c6f4dbfd03f5e672675d97919ecd/src/OpenLoco/src/Station.cpp#L424

There was a line in `Station.cpp` that said the catchment flags shouldn't be shifted. Is this the correct way to implement these?

Comment (ZehMatt) [reply src/OpenLoco/src/Station.h:92]: This specific enum doesn't hold flags, you can remove OPENLOCO_ENABLE_ENUM_OPERATORS but keep it enum class.

File: src/OpenLoco/src/Station.cpp [hunk 2]
```diff
@@ -244,11 +244,11 @@ namespace OpenLoco
 
         cargoSearchState.resetIndustryMap();
 
-        setCatchmentDisplay(1);
+        setCatchmentDisplay(CatchmentFlags::flag1);
 
         if (location.x != -1)
         {
-            sub_491BF5(location, 1);
+            sub_491BF5(location, CatchmentFlags::flag0);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/Station.cpp:251]: Changed value.

File: src/OpenLoco/src/Windows/StationWindow.cpp [hunk 3]
```diff
@@ -834,10 +834,13 @@ namespace OpenLoco::Ui::Windows::Station
                 StringIds::label_icons_road_air_water,
                 StringIds::label_icons_rail_road_air_water
             };
+
+            uint16_t stationType = static_cast<std::underlying_type_t<StationFlags>>(station->flags) & 0xF;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/StationWindow.cpp:838]: Should perhaps be a member function of the Station struct.

File: src/OpenLoco/src/World/Station.cpp [hunk 4]
```diff
@@ -54,17 +54,17 @@ namespace OpenLoco
             _map[y * kMapColumns + x] &= ~(1 << 1);
         }
 
-        void setTile(const tile_coord_t x, const tile_coord_t y, const uint8_t flag)
+        void setTile(const tile_coord_t x, const tile_coord_t y, const CatchmentFlags flag)
         {
-            _map[y * kMapColumns + x] |= (1 << flag);
+            _map[y * kMapColumns + x] |= (1 << static_cast<std::underlying_type_t<CatchmentFlags>>(flag));
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/Station.cpp:59]: enumValue

File: src/OpenLoco/src/World/Station.cpp [hunk 5]
```diff
@@ -54,17 +54,17 @@ namespace OpenLoco
...
         }
 
-        void resetTile(const tile_coord_t x, const tile_coord_t y, const uint8_t flag)
+        void resetTile(const tile_coord_t x, const tile_coord_t y, const CatchmentFlags flag)
         {
-            _map[y * kMapColumns + x] &= ~(1 << flag);
+            _map[y * kMapColumns + x] &= ~(1 << static_cast<std::underlying_type_t<CatchmentFlags>>(flag));
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/Station.cpp:64]: enumValue

File: src/OpenLoco/src/World/Station.cpp [hunk 6]
```diff
@@ -851,6 +851,8 @@ namespace OpenLoco
             StringIds::label_icons_road_air_water,
             StringIds::label_icons_rail_road_air_water,
         };
-        return label_icons[flags & StationFlags::allModes];
+
+        uint16_t label = static_cast<std::underlying_type_t<StationFlags>>(flags & StationFlags::allModes);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/World/Station.cpp:855]: enumValue

**Review by duncanspumpkin — APPROVED** [review]
I don't like the CatchmentFlags but its outside the scope of this PR.

**Pull request merged**